### PR TITLE
feat: migrate transactions to sql database

### PR DESCRIPTION
**Pull request by nicoloboschi**
* Use transactions table on sqldb instead of duckdb
* Transactions are wrote best-effort async to avoid blocking hot path 
* Ensure migrations happen at the startup
* Added file lock before acquiring duckdb connection to avoid concurrency errors
* Fixed a bug whereas all transactions and built vertex are wiped out at every startup
* Improved conccurreny in services

### CODE REVIEW

File: src/backend/base/langflow/graph/graph/base.py [hunk 1]
```diff
@@ -914,13 +914,13 @@ async def build_vertex(
             else:
                 raise ValueError(f"No result found for vertex {vertex_id}")
             flow_id = self.flow_id
-            log_transaction(flow_id, vertex, status="success")
+            asyncio.create_task(log_transaction(flow_id, vertex, status="success"))
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/graph/graph/base.py:917]: You should probably remove these two calls here. I feel like these should be vertex builds instead, and we can add a tab in the logs table later.

File: src/backend/base/langflow/graph/graph/base.py [hunk 2]
```diff
@@ -914,13 +914,13 @@ async def build_vertex(
...
             return result_dict, params, valid, artifacts, vertex
         except Exception as exc:
             if not isinstance(exc, ComponentBuildException):
                 logger.exception(f"Error building Component: \n\n{exc}")
             flow_id = self.flow_id
-            log_transaction(flow_id, vertex, status="failure", error=str(exc))
+            asyncio.create_task(log_transaction(flow_id, vertex, status="failure", error=str(exc)))
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/graph/graph/base.py:923]: This one as well.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @nicoloboschi 

Thank you for this. 

Just some minor comments and this should be good to go.

File: src/backend/base/langflow/graph/graph/base.py (hunk 1 above)
Comment (nicoloboschi) [reply src/backend/base/langflow/graph/graph/base.py]: removed, thanks

File: src/backend/base/langflow/graph/graph/base.py (hunk 2 above)
Comment (nicoloboschi) [reply src/backend/base/langflow/graph/graph/base.py]: removed, thanks

**Review by zzzming — APPROVED** [review]

**Review by ogabrielluiz — APPROVED** [review]

**Pull request merged**